File: src/test/java/test/ListRoleDepartmentsTest.java
package test;
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class ListRoleDepartmentsTest {

    private static final String ACCESS_KEY_ID = "AUTHING_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "AUTHING_ACCESS_KEY_SECRET";

    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions();
        clientOptions.setAccessKeyId(ACCESS_KEY_ID);
        clientOptions.setAccessKeySecret(ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        ListRoleDepartmentsDto request = new ListRoleDepartmentsDto();
        request.setCode("code_9934");
        request.setNamespace("namespace_367");
        request.setPage(0);
        request.setLimit(0);

        RoleDepartmentListPaginatedRespDto response = client.listRoleDepartments(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/ListRoleDepartmentsTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.ListRoleDepartmentsDto{
        Code: "code_6936",
        Namespace: "namespace_6096",
        Page: 0,
        Limit: 0,
    }
    response := client.ListRoleDepartments(&request)
    fmt.Println(response)
}

